src/MongoDB.Driver.Core/Core/Misc/ServerVersion.cs

```diff
@@ -0,0 +1,505 @@
+        /// <returns>True if the string representation was parsed successfully; otherwise false.</returns>
+        public static bool TryParse(string value, out ServerVersion result)
+        {
+            if (!string.IsNullOrEmpty(value))
+            {
+                var pattern = @"(?<major>\d+)\.(?<minor>\d+)(\.(?<patch>\d+)(-(?<preRelease>.*))?)?";
+                var match = Regex.Match((string)value, pattern);
+                if (match.Success)
+                {
+                    var major = int.Parse(match.Groups["major"].Value);
+                    var minor = int.Parse(match.Groups["minor"].Value);
+                    var patch = match.Groups["patch"].Success ? int.Parse(match.Groups["patch"].Value) : 0;
+                    var preRelease = match.Groups["preRelease"].Success ? match.Groups["preRelease"].Value : null;
+
+                    result = new ServerVersion(major, minor, patch, preRelease);
+                    return true;
+                }
+            }
+
+            result = null;
+            return false;
+        }
+
+        // public operators
+        /// <summary>
+        /// Casts a SemanticVersion to a ServerVersion.
+        /// </summary>
+        /// <param name="semanticVersion">The SemanticVersion.</param>
+        /// <returns>A ServerVersion.</returns>
+        /// <exception cref="FormatException">semanticVersion</exception>
+        public static explicit operator ServerVersion(SemanticVersion semanticVersion)
+        {
+            return Parse(semanticVersion.ToString());
+        }
+
+        /// <summary>
+        /// Determines whether two specified server versions have the same value.
+        /// </summary>
+        /// <param name="a">The first server version to compare, or null.</param>
+        /// <param name="b">The second server version to compare, or null.</param>
+        /// <returns>
+        /// True if the value of a is the same as the value of b; otherwise false.
+        /// </returns>
+        public static bool operator ==(ServerVersion a, ServerVersion b)
+        {
+            if (ReferenceEquals(a, null))
+            {
+                return ReferenceEquals(b, null);
+            }
+
+            return a.CompareTo(b) == 0;
+        }
+
+        /// <summary>
+        /// Determines whether two specified server versions have different values.
+        /// </summary>
+        /// <param name="a">The first server version to compare, or null.</param>
+        /// <param name="b">The second server version to compare, or null.</param>
+        /// <returns>
+        /// True if the value of a is different from the value of b; otherwise false.
+        /// </returns>
+        public static bool operator !=(ServerVersion a, ServerVersion b)
+        {
+            if (ReferenceEquals(a, null))
+            {
+                return !ReferenceEquals(b, null);
+            }
+
+            return a.CompareTo(b) != 0;
```

Comment:
Make this and all comparison operators match the corresponding implementations in SemanticServer.cs (and see comments in SemanticServer.cs regarding these).

Reply:
I explained my point in `SemanticVersion`.
Not done.

Reply:
Done.